### PR TITLE
kunlunxin berfLarge inference configs && results

### PR DESCRIPTION
**Pull request by jinxiangshi**
**Results:**

2023-08-24 08:29:09.106 | Finish Info | __main__:<module>:159 - {'vendor': 'kunlunxin', 'compiler': 'xtcl', 'precision': 'fp32', 'batchsize': 32, 'flops': 18923182080000.0, 'e2e_time(second)': 6102.158, 'p_validation_whole(qps)': None, 'p_validation_core(qps)': None, 'p_inference_whole(qps)': 49.631, '*p_inference_core(qps)': 55.999, 'val_average_acc': None, 'infer_average_acc': 0.638}

### CODE REVIEW

File: inference/docker_images/kunlunxin/pytorch_1.13/Dockerfile [hunk 1]
```diff
@@ -46,7 +46,8 @@ RUN /root/miniconda/envs/python38/bin/pip install \
     munch \
     pyyaml \
     tqdm \
-    scipy
+    scipy \
+    transformers
```
Comment (shh2000) [inline inference/docker_images/kunlunxin/pytorch_1.13/Dockerfile:50]: bert的requirements中已有transformers，这属于是bert这个case而非kunlunxin硬件或编译器需要的。请删除这一改动看看是否会遇到运行错误？

Comment (jinxiangshi) [reply inference/docker_images/kunlunxin/pytorch_1.13/Dockerfile]: 确认了下，不需要重复 `install transformers` 。新的patch去掉了这处改动

**Review by shh2000 — APPROVED** [review]

**Comment by shh2000** [comment]
请将运行log邮件发给我（13820618441@163.com）或上传到这个PR中吧，我们存档一下

**Comment by jinxiangshi** [comment]
Downloading (…)solve/main/vocab.txt: 100%|██████████| 232k/232k [00:00<00:00, 1.82MB/s]
Downloading (…)okenizer_config.json: 100%|██████████| 28.0/28.0 [00:00<00:00, 24.4kB/s]
Downloading (…)lve/main/config.json: 100%|██████████| 571/571 [00:00<00:00, 528kB/s]
Downloading model.safetensors: 100%|██████████| 1.34G/1.34G [06:59<00:00, 3.21MB/s]
Some weights of the model checkpoint at bert-large-uncased were not used when initializing BertForMaskedLM: ['cls.seq_relationship.bias', 'cls.seq_relationship.weight', 'bert.pooler.dense.weight', 'bert.pooler.dense.bias']
- This IS expected if you are initializing BertForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
2023-08-24 06:56:17.497 | Init End | __main__:main:33 - 530.132958650589 seconds
2023-08-24 06:56:17.498 | Model Forward Begin | __main__:main:37 -
2023-08-24 06:56:17.498 | Model Forward End | __main__:main:45 -
2023-08-24 06:56:17.498 | Export Begin | __main__:main:51 - Export pytorch model into .onnx
2023-08-24 06:56:17.498 | Export End | __main__:main:58 - 3.814697265625e-06 seconds
XPURT /root/XTCL-ubuntu_x86_64/runtime/shlib/libxpurt.so.1 loaded
2023-08-24 06:56:19.105 | Vendor Compile Begin | __main__:main:66 - Compiling With kunlunxin.xtcl
2023-08-24 06:56:59.869 | Vendor Compile End | __main__:main:73 - 40.76388502120972 seconds
2023-08-24 06:56:59.870 | Vendor Inference Begin | __main__:main:77 -
2023-08-24 08:29:09.013 | INFO     | benchmarks.bertLarge.pytorch.forward:engine_forward:107 - MaskedLM Acc: 0.6381359052507397
2023-08-24 08:29:09.013 | INFO     | benchmarks.bertLarge.pytorch.forward:cal_perf:10 - Inference(pytorch) Perf: 49.630810177950245 qps
2023-08-24 08:29:09.013 | INFO     | benchmarks.bertLarge.pytorch.forward:cal_perf:13 - Inference(pytorch) core Perf: 55.998627583778045 qps
2023-08-24 08:29:09.021 | Vendor Inference End | __main__:main:83 -
2023-08-24 08:29:09.106 | Finish Info | __main__:<module>:159 - {'vendor': 'kunlunxin', 'compiler': 'xtcl', 'precision': 'fp32', 'batchsize': 32, 'flops': 18923182080000.0, 'e2e_time(second)': 6102.158, 'p_validation_whole(qps)': None, 'p_validation_core(qps)': None, 'p_inference_whole(qps)': 49.631, '*p_inference_core(qps)': 55.999, 'val_average_acc': None, 'infer_average_acc': 0.638}

**Comment by jinxiangshi** [comment]
收到。
我新提了patch，并且把log放在了pr里



---Original---
From: ***@***.***&gt;
Date: Fri, Aug 25, 2023 13:41 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [FlagOpen/FlagPerf] kunlunxin berfLarge inference configs &amp;&amp;results (PR #212)




 
***@***.***）或上传到这个PR中吧，我们存档一下

**Comment by jinxiangshi** [comment]
2023-08-28 22:32:19.855 | INFO     | __main__:main:616 - -== Testcase bertLarge:pytorch_1.13 finished ==-
2023-08-28 22:32:19.855 | INFO     | __main__:main:617 - === 2.2 Setup container and run testcases finished. ===
2023-08-28 22:32:19.855 | INFO     | __main__:main:619 - === 2.3 Compilation Case Performance ===
2023-08-28 22:32:19.860 | INFO     | __main__:compilation_result:461 - vendor                                      : kunlunxin
2023-08-28 22:32:19.860 | INFO     | __main__:compilation_result:461 - compiler                                    : xtcl
2023-08-28 22:32:19.861 | INFO     | __main__:compilation_result:461 - precision                                   : fp32
2023-08-28 22:32:19.861 | INFO     | __main__:compilation_result:461 - batchsize                                   : 32
2023-08-28 22:32:19.861 | INFO     | __main__:compilation_result:461 - flops                                       : 42220738560000.0
2023-08-28 22:32:19.861 | INFO     | __main__:compilation_result:461 - e2e_time(second)                            : 3867.652
2023-08-28 22:32:19.861 | INFO     | __main__:compilation_result:461 - p_validation_whole(qps)                     : None
2023-08-28 22:32:19.862 | INFO     | __main__:compilation_result:461 - p_validation_core(qps)                      : None
2023-08-28 22:32:19.862 | INFO     | __main__:compilation_result:461 - p_inference_whole(qps)                      : 93.851
2023-08-28 22:32:19.862 | INFO     | __main__:compilation_result:461 - *p_inference_core(qps)                      : 124.943
2023-08-28 22:32:19.862 | INFO     | __main__:compilation_result:461 - val_average_acc                             : None
2023-08-28 22:32:19.862 | INFO     | __main__:compilation_result:461 - infer_average_acc                           : 0.638
2023-08-28 22:32:19.863 | INFO     | __main__:compilation_result:461 - vendor_usage(GiB)                           : 0.0
2023-08-28 22:32:19.863 | INFO     | __main__:compilation_result:461 - vendor_max_mem(GiB)                         : 16.0
2023-08-28 22:32:19.863 | INFO     | __main__:compilation_result:461 - *MFU                                        : 131.9%

**Comment by shh2000** [comment]
> 2023-08-28 22:32:19.855 | INFO | **main**:main:616 - -== Testcase bertLarge:pytorch_1.13 finished ==- 2023-08-28 22:32:19.855 | INFO | **main**:main:617 - === 2.2 Setup container and run testcases finished. === 2023-08-28 22:32:19.855 | INFO | **main**:main:619 - === 2.3 Compilation Case Performance === 2023-08-28 22:32:19.860 | INFO | **main**:compilation_result:461 - vendor : kunlunxin 2023-08-28 22:32:19.860 | INFO | **main**:compilation_result:461 - compiler : xtcl 2023-08-28 22:32:19.861 | INFO | **main**:compilation_result:461 - precision : fp32 2023-08-28 22:32:19.861 | INFO | **main**:compilation_result:461 - batchsize : 32 2023-08-28 22:32:19.861 | INFO | **main**:compilation_result:461 - flops : 42220738560000.0 2023-08-28 22:32:19.861 | INFO | **main**:compilation_result:461 - e2e_time(second) : 3867.652 2023-08-28 22:32:19.861 | INFO | **main**:compilation_result:461 - p_validation_whole(qps) : None 2023-08-28 22:32:19.862 | INFO | **main**:compilation_result:461 - p_validation_core(qps) : None 2023-08-28 22:32:19.862 | INFO | **main**:compilation_result:461 - p_inference_whole(qps) : 93.851 2023-08-28 22:32:19.862 | INFO | **main**:compilation_result:461 - *p_inference_core(qps) : 124.943 2023-08-28 22:32:19.862 | INFO | **main**:compilation_result:461 - val_average_acc : None 2023-08-28 22:32:19.862 | INFO | **main**:compilation_result:461 - infer_average_acc : 0.638 2023-08-28 22:32:19.863 | INFO | **main**:compilation_result:461 - vendor_usage(GiB) : 0.0 2023-08-28 22:32:19.863 | INFO | **main**:compilation_result:461 - vendor_max_mem(GiB) : 16.0 2023-08-28 22:32:19.863 | INFO | **main**:compilation_result:461 - *MFU : 131.9%

这里有两个问题：
1. MFU已经超过了1.0，这是一个比例值。请看一下运行时的权重/激活是否真的为float32，并且查看一下kunlunxin_analysis中写的理论算力是否正确
2. 显存使用为0，请检查kunlunxin_monitor.py写的是否正确，或kunlunxin_analysis中计算的最大显存使用量是否正确

File: inference/benchmarks/bertLarge/README.md [hunk 2]
```diff
@@ -64,5 +82,5 @@ bert_reference_results_text_md5.txt
 | ----------- | --------- | ---- | ---- | -------- | ----------- | ---------- | ------------- | ------------ | ----------- | ----------- |
 | tensorrt | fp16      | 32 | 1283.9   | 257.3       | 260.4      | 408.3         | 418.1          | 45.3% | 0.600/0.638 | 17.4/40.0 |
 | tensorrt | fp32   | 32 | 1868.8   | 150.4       | 152.2      | 190.4         | 194.1       | 42.0% | 0.638/0.638 | 16.9/40.0 |
-
+| kunlunxin_xtcl| fp32   | 32 |3867.6 | None          | None       | 93.8          | 124.9          | 131.9% | 0.638/0.638| 16.0/32.0|
```
Comment (shh2000) [inline inference/benchmarks/bertLarge/README.md:85]: precision=W32A16

**Review by upvenly — APPROVED** [review]

**Pull request merged**